From 6bddeff25157fd03cf0c9b6fef3a9b0a7187829f Mon Sep 17 00:00:00 2001
From: Andy CrossGate Yan <GeForce8800Ultra@gmail.com>
Date: Sun, 30 Jun 2024 15:53:25 +0800
Subject: [PATCH] Revert "Adding ColorUpdateLogger to track color updates so
 that we demonstrate the root cause of our theme bugs"

This reverts commit 3a819dee1225de25f0e228ef2c1b9447eb081661.

Change-Id: I7f58b1fa44339d3cf76d5254c7b3aceb66e07c10

[Port to A16] Resolved via 3-way merge + manual conflict resolution: kept A16 refactors (shelf alignment, footer blur, notif transparency flags), stripped ColorUpdateLogger/DrawableDump and the aconfig flag. KEPT getNightModeName (A16 ConfigurationControllerDelegate depends on it). Tests at A16 locations (multivalentTests, tests/utils) cleaned; A14-only test paths dropped.

Change-Id: I638e33a05dccd1987bfbdf3ad8be3ec28674fc26
---
 packages/SystemUI/aconfig/systemui.aconfig    |   7 -
 ...ExpandableNotificationRowControllerTest.kt |   3 -
 ...cationStackScrollLayoutControllerTest.java |   3 -
 .../systemui/statusbar/NotificationShelf.java |  18 +-
 .../notification/ColorUpdateLogger.kt         | 153 -------------
 .../coordinator/ViewConfigCoordinator.kt      |  18 +-
 .../footer/ui/view/FooterView.java            |  20 --
 .../row/ActivatableNotificationView.java      |   6 +-
 .../row/ExpandableNotificationRow.java        |  21 +-
 .../ExpandableNotificationRowController.java  |   5 -
 .../notification/row/ExpandableView.java      |   5 +-
 .../row/NotificationBackgroundView.java       |  13 +-
 .../stack/NotificationStackScrollLayout.java  |  12 +-
 ...tificationStackScrollLayoutController.java |  10 -
 .../com/android/systemui/util/DrawableDump.kt | 201 ------------------
 .../coordinator/ViewConfigCoordinatorTest.kt  |   3 -
 .../row/ExpandableNotificationRowBuilder.kt   |   2 -
 17 files changed, 13 insertions(+), 487 deletions(-)
 delete mode 100644 packages/SystemUI/src/com/android/systemui/statusbar/notification/ColorUpdateLogger.kt
 delete mode 100644 packages/SystemUI/src/com/android/systemui/util/DrawableDump.kt

diff --git a/packages/SystemUI/aconfig/systemui.aconfig b/packages/SystemUI/aconfig/systemui.aconfig
index 28ecabe50d68..12c11b0b7f13 100644
--- a/packages/SystemUI/aconfig/systemui.aconfig
+++ b/packages/SystemUI/aconfig/systemui.aconfig
@@ -111,13 +111,6 @@ flag {
     bug: "217799515"
 }
 
-flag {
-    name: "notification_color_update_logger"
-    namespace: "systemui"
-    description: "Enabled debug logging and dumping of notification color updates."
-    bug: "294347738"
-}
-
 flag {
     name: "notifications_icon_container_refactor"
     namespace: "systemui"
diff --git a/packages/SystemUI/multivalentTests/src/com/android/systemui/statusbar/notification/row/ExpandableNotificationRowControllerTest.kt b/packages/SystemUI/multivalentTests/src/com/android/systemui/statusbar/notification/row/ExpandableNotificationRowControllerTest.kt
index 36806fbc0468..e417724ef74e 100644
--- a/packages/SystemUI/multivalentTests/src/com/android/systemui/statusbar/notification/row/ExpandableNotificationRowControllerTest.kt
+++ b/packages/SystemUI/multivalentTests/src/com/android/systemui/statusbar/notification/row/ExpandableNotificationRowControllerTest.kt
@@ -35,7 +35,6 @@ import com.android.systemui.plugins.PluginManager
 import com.android.systemui.plugins.statusbar.StatusBarStateController
 import com.android.systemui.statusbar.SmartReplyController
 import com.android.systemui.statusbar.notification.BundleInteractionLogger
-import com.android.systemui.statusbar.notification.ColorUpdateLogger
 import com.android.systemui.statusbar.notification.NotificationActivityStarter
 import com.android.systemui.statusbar.notification.collection.EntryAdapter
 import com.android.systemui.statusbar.notification.collection.EntryAdapterFactory
@@ -88,7 +87,6 @@ class ExpandableNotificationRowControllerTest : SysuiTestCase() {
     private val rivSubComponentFactory: RemoteInputViewSubcomponent.Factory = mock()
     private val metricsLogger: MetricsLogger = mock()
     private val logBufferLogger = NotificationRowLogger(logcatLogBuffer(), logcatLogBuffer())
-    private val colorUpdateLogger: ColorUpdateLogger = mock()
     private val listContainer: NotificationListContainer = mock()
     private val smartReplyConstants: SmartReplyConstants = mock()
     private val smartReplyController: SmartReplyController = mock()
@@ -136,7 +134,6 @@ class ExpandableNotificationRowControllerTest : SysuiTestCase() {
             activableNotificationViewController,
             rivSubComponentFactory,
             metricsLogger,
-            colorUpdateLogger,
             logBufferLogger,
             NotificationChildrenContainerLogger(logcatLogBuffer()),
             listContainer,
diff --git a/packages/SystemUI/multivalentTests/src/com/android/systemui/statusbar/notification/stack/NotificationStackScrollLayoutControllerTest.java b/packages/SystemUI/multivalentTests/src/com/android/systemui/statusbar/notification/stack/NotificationStackScrollLayoutControllerTest.java
index 5fbf8bf0bea5..db736139c16d 100644
--- a/packages/SystemUI/multivalentTests/src/com/android/systemui/statusbar/notification/stack/NotificationStackScrollLayoutControllerTest.java
+++ b/packages/SystemUI/multivalentTests/src/com/android/systemui/statusbar/notification/stack/NotificationStackScrollLayoutControllerTest.java
@@ -73,7 +73,6 @@ import com.android.systemui.statusbar.LockscreenShadeTransitionController;
 import com.android.systemui.statusbar.NotificationLockscreenUserManager;
 import com.android.systemui.statusbar.NotificationLockscreenUserManager.UserChangedListener;
 import com.android.systemui.statusbar.SysuiStatusBarStateController;
-import com.android.systemui.statusbar.notification.ColorUpdateLogger;
 import com.android.systemui.statusbar.notification.DynamicPrivacyController;
 import com.android.systemui.statusbar.notification.NotificationWakeUpCoordinator;
 import com.android.systemui.statusbar.notification.collection.NotifCollection;
@@ -139,7 +138,6 @@ public class NotificationStackScrollLayoutControllerTest extends SysuiTestCase {
     @Mock private PowerInteractor mPowerInteractor;
     @Mock private NotificationLockscreenUserManager mNotificationLockscreenUserManager;
     @Mock private MetricsLogger mMetricsLogger;
-    @Mock private ColorUpdateLogger mColorUpdateLogger;
     @Mock private DumpManager mDumpManager;
     @Mock(answer = Answers.RETURNS_SELF)
     private NotificationSwipeHelper.Builder mNotificationSwipeHelperBuilder;
@@ -842,7 +840,6 @@ public class NotificationStackScrollLayoutControllerTest extends SysuiTestCase {
                 mPowerInteractor,
                 mNotificationLockscreenUserManager,
                 mMetricsLogger,
-                mColorUpdateLogger,
                 mDumpManager,
                 new FalsingCollectorFake(),
                 new FalsingManagerFake(),
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/NotificationShelf.java b/packages/SystemUI/src/com/android/systemui/statusbar/NotificationShelf.java
index fd20a01b7e2b..6d31bdc9660b 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/NotificationShelf.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/NotificationShelf.java
@@ -19,7 +19,6 @@ package com.android.systemui.statusbar;
 import static com.android.keyguard.BouncerPanelExpansionCalculator.aboutToShowBouncerProgress;
 import static com.android.systemui.Flags.notificationRowTransparency;
 import static com.android.systemui.Flags.physicalNotificationMovement;
-import static com.android.systemui.util.ColorUtilKt.hexColorString;
 
 import android.content.Context;
 import android.content.res.Configuration;
@@ -46,7 +45,6 @@ import com.android.systemui.animation.ShadeInterpolation;
 import com.android.systemui.res.R;
 import com.android.systemui.scene.shared.flag.SceneContainerFlag;
 import com.android.systemui.shade.transition.LargeScreenShadeInterpolator;
-import com.android.systemui.statusbar.notification.ColorUpdateLogger;
 import com.android.systemui.statusbar.notification.NotificationUtils;
 import com.android.systemui.statusbar.notification.SourceType;
 import com.android.systemui.statusbar.notification.row.ActivatableNotificationView;
@@ -196,8 +194,8 @@ public class NotificationShelf extends ActivatableNotificationView {
 
     @Override
     public String toString() {
-        return super.toString()
-                + " (hideBackground=" + mHideBackground
+        return "NotificationShelf"
+                + "(hideBackground=" + mHideBackground
                 + " notGoneIndex=" + mNotGoneIndex
                 + " hasItemsInStableShelf=" + mHasItemsInStableShelf
                 + " interactive=" + mInteractive
@@ -435,18 +433,6 @@ public class NotificationShelf extends ActivatableNotificationView {
         }
     }
 
-    @Override
-    public void updateBackgroundColors() {
-        super.updateBackgroundColors();
-        ColorUpdateLogger colorUpdateLogger = ColorUpdateLogger.getInstance();
-
-        if (colorUpdateLogger != null) {
-            colorUpdateLogger.logEvent("Shelf.updateBackgroundColors()",
-                    "normalBgColor=" + hexColorString(getNormalBgColor())
-                            + " background=" + mBackgroundNormal.toDumpString());
-        }
-    }
-
     /**
      * Update the shelf appearance based on the other notifications around it. This transforms
      * the icons from the notification area into the shelf.
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/notification/ColorUpdateLogger.kt b/packages/SystemUI/src/com/android/systemui/statusbar/notification/ColorUpdateLogger.kt
deleted file mode 100644
index 0f0ab2e36b8d..000000000000
--- a/packages/SystemUI/src/com/android/systemui/statusbar/notification/ColorUpdateLogger.kt
+++ /dev/null
@@ -1,153 +0,0 @@
-/*
- * Copyright (C) 2023 The Android Open Source Project
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License
- */
-
-package com.android.systemui.statusbar.notification
-
-import android.icu.text.SimpleDateFormat
-import android.util.IndentingPrintWriter
-import com.android.systemui.Dumpable
-import com.android.systemui.Flags.notificationColorUpdateLogger
-import com.android.systemui.dagger.SysUISingleton
-import com.android.systemui.dump.DumpManager
-import com.android.systemui.flags.FeatureFlagsClassic
-import com.android.systemui.util.Compile
-import com.android.systemui.util.asIndenting
-import com.android.systemui.util.printCollection
-import com.android.systemui.util.withIncreasedIndent
-import com.google.errorprone.annotations.CompileTimeConstant
-import java.io.PrintWriter
-import java.util.Locale
-import java.util.SortedSet
-import java.util.TreeSet
-import javax.inject.Inject
-
-@SysUISingleton
-class ColorUpdateLogger
-@Inject
-constructor(
-    val featureFlags: FeatureFlagsClassic,
-    dumpManager: DumpManager,
-) : Dumpable {
-
-    inline val isEnabled
-        get() = Compile.IS_DEBUG && notificationColorUpdateLogger()
-    private val frames: MutableList<Frame> = mutableListOf()
-
-    init {
-        dumpManager.registerDumpable(this)
-        if (isEnabled) {
-            instance = this
-        }
-    }
-
-    @JvmOverloads
-    fun logTriggerEvent(@CompileTimeConstant type: String, extra: String? = null) {
-        if (!isEnabled) return
-        val event = Event(type = type, extraValue = extra)
-        val didAppend = frames.lastOrNull()?.tryAddTrigger(event) == true
-        if (!didAppend) {
-            frames.add(Frame(event))
-            if (frames.size > maxFrames) frames.removeAt(0)
-        }
-    }
-
-    @JvmOverloads
-    fun logEvent(@CompileTimeConstant type: String, extra: String? = null) {
-        if (!isEnabled) return
-        val frame = frames.lastOrNull() ?: return
-        frame.events.add(Event(type = type, extraValue = extra))
-        frame.trim()
-    }
-
-    @JvmOverloads
-    fun logNotificationEvent(
-        @CompileTimeConstant type: String,
-        key: String,
-        extra: String? = null
-    ) {
-        if (!isEnabled) return
-        val frame = frames.lastOrNull() ?: return
-        frame.events.add(Event(type = type, extraValue = extra, notificationKey = key))
-        frame.trim()
-    }
-
-    override fun dump(pwOrig: PrintWriter, args: Array<out String>) {
-        val pw = pwOrig.asIndenting()
-        pw.println("enabled: $isEnabled")
-        pw.printCollection("frames", frames) { it.dump(pw) }
-    }
-
-    private class Frame(event: Event) {
-        val startTime: Long = event.time
-        val events: MutableList<Event> = mutableListOf(event)
-        val triggers: SortedSet<String> = TreeSet<String>().apply { add(event.type) }
-        var trimmedEvents: Int = 0
-
-        fun tryAddTrigger(newEvent: Event): Boolean {
-            if (newEvent.time < startTime) return false
-            if (newEvent.time - startTime > triggerStartsNewFrameAge) return false
-            if (newEvent.type in triggers) return false
-            triggers.add(newEvent.type)
-            events.add(newEvent)
-            trim()
-            return true
-        }
-
-        fun trim() {
-            if (events.size > maxEventsPerFrame) {
-                events.removeAt(0)
-                trimmedEvents++
-            }
-        }
-
-        fun dump(pw: IndentingPrintWriter) {
-            pw.println("Frame")
-            pw.withIncreasedIndent {
-                pw.println("startTime: ${timeString(startTime)}")
-                pw.printCollection("triggers", triggers)
-                pw.println("trimmedEvents: $trimmedEvents")
-                pw.printCollection("events", events) { it.dump(pw) }
-            }
-        }
-    }
-
-    private class Event(
-        @CompileTimeConstant val type: String,
-        val extraValue: String? = null,
-        val notificationKey: String? = null,
-    ) {
-        val time: Long = System.currentTimeMillis()
-
-        fun dump(pw: IndentingPrintWriter) {
-            pw.append(timeString(time)).append(": ").append(type)
-            extraValue?.let { pw.append(" ").append(it) }
-            notificationKey?.let { pw.append(" ---- ").append(logKey(it)) }
-            pw.println()
-        }
-    }
-
-    private companion object {
-        @JvmStatic
-        var instance: ColorUpdateLogger? = null
-            private set
-        private const val maxFrames = 5
-        private const val maxEventsPerFrame = 250
-        private const val triggerStartsNewFrameAge = 5000
-
-        private val dateFormat = SimpleDateFormat("MM-dd HH:mm:ss.SSS", Locale.US)
-        private fun timeString(time: Long): String = dateFormat.format(time)
-    }
-}
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/notification/collection/coordinator/ViewConfigCoordinator.kt b/packages/SystemUI/src/com/android/systemui/statusbar/notification/collection/coordinator/ViewConfigCoordinator.kt
index 9785d54207e4..6bc09e941029 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/notification/collection/coordinator/ViewConfigCoordinator.kt
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/notification/collection/coordinator/ViewConfigCoordinator.kt
@@ -25,7 +25,6 @@ import com.android.keyguard.KeyguardUpdateMonitorCallback
 import com.android.systemui.shade.ShadeDisplayAware
 import com.android.systemui.statusbar.NotificationLockscreenUserManager
 import com.android.systemui.statusbar.NotificationLockscreenUserManager.UserChangedListener
-import com.android.systemui.statusbar.notification.ColorUpdateLogger
 import com.android.systemui.statusbar.notification.collection.NotifPipeline
 import com.android.systemui.statusbar.notification.collection.coordinator.dagger.CoordinatorScope
 import com.android.systemui.statusbar.notification.row.NotificationGutsManager
@@ -45,8 +44,7 @@ internal constructor(
     @ShadeDisplayAware private val mConfigurationController: ConfigurationController,
     private val mLockscreenUserManager: NotificationLockscreenUserManager,
     private val mGutsManager: NotificationGutsManager,
-    private val mKeyguardUpdateMonitor: KeyguardUpdateMonitor,
-    private val colorUpdateLogger: ColorUpdateLogger,
+    private val mKeyguardUpdateMonitor: KeyguardUpdateMonitor
 ) : Coordinator, ConfigurationController.ConfigurationListener {
 
     private var mIsSwitchingUser = false
@@ -57,15 +55,11 @@ internal constructor(
     private val mKeyguardUpdateCallback =
         object : KeyguardUpdateMonitorCallback() {
             override fun onUserSwitching(userId: Int) {
-                colorUpdateLogger.logTriggerEvent("VCC.mKeyguardUpdateCallback.onUserSwitching()")
                 log { "ViewConfigCoordinator.onUserSwitching(userId=$userId)" }
                 mIsSwitchingUser = true
             }
 
             override fun onUserSwitchComplete(userId: Int) {
-                colorUpdateLogger.logTriggerEvent(
-                    "VCC.mKeyguardUpdateCallback.onUserSwitchComplete()"
-                )
                 log { "ViewConfigCoordinator.onUserSwitchComplete(userId=$userId)" }
                 mIsSwitchingUser = false
                 applyChangesOnUserSwitched()
@@ -75,7 +69,6 @@ internal constructor(
     private val mUserChangedListener =
         object : UserChangedListener {
             override fun onUserChanged(userId: Int) {
-                colorUpdateLogger.logTriggerEvent("VCC.mUserChangedListener.onUserChanged()")
                 log { "ViewConfigCoordinator.onUserChanged(userId=$userId)" }
                 applyChangesOnUserSwitched()
             }
@@ -89,7 +82,6 @@ internal constructor(
     }
 
     override fun onDensityOrFontScaleChanged() {
-        colorUpdateLogger.logTriggerEvent("VCC.onDensityOrFontScaleChanged()")
         log {
             val keyguardIsSwitchingUser = mKeyguardUpdateMonitor.isSwitchingUser
             "ViewConfigCoordinator.onDensityOrFontScaleChanged()" +
@@ -106,7 +98,6 @@ internal constructor(
     }
 
     override fun onUiModeChanged() {
-        colorUpdateLogger.logTriggerEvent("VCC.onUiModeChanged()")
         log {
             val keyguardIsSwitchingUser = mKeyguardUpdateMonitor.isSwitchingUser
             "ViewConfigCoordinator.onUiModeChanged()" +
@@ -121,12 +112,10 @@ internal constructor(
     }
 
     override fun onThemeChanged() {
-        colorUpdateLogger.logTriggerEvent("VCC.onThemeChanged()")
         onDensityOrFontScaleChanged()
     }
 
     private fun applyChangesOnUserSwitched() {
-        colorUpdateLogger.logEvent("VCC.applyChangesOnUserSwitched()")
         if (mReinflateNotificationsOnUserSwitched) {
             updateNotificationsOnDensityOrFontScaleChanged()
             mReinflateNotificationsOnUserSwitched = false
@@ -138,10 +127,6 @@ internal constructor(
     }
 
     private fun updateNotificationsOnUiModeChanged() {
-        colorUpdateLogger.logEvent(
-            "VCC.updateNotificationsOnUiModeChanged()",
-            "mode=" + mConfigurationController.nightModeName,
-        )
         log { "ViewConfigCoordinator.updateNotificationsOnUiModeChanged()" }
         traceSection("updateNotifOnUiModeChanged") {
             mPipeline?.allNotifs?.forEach { entry ->
@@ -153,7 +138,6 @@ internal constructor(
     }
 
     private fun updateNotificationsOnDensityOrFontScaleChanged() {
-        colorUpdateLogger.logEvent("VCC.updateNotificationsOnDensityOrFontScaleChanged()")
         mPipeline?.allNotifs?.forEach { entry -> entry.onDensityOrFontScaleChanged() }
         mPipeline?.allBundles?.forEach { entry -> entry.onDensityOrFontScaleChanged() }
         mGutsManager.closeAndUndoGuts()
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/notification/footer/ui/view/FooterView.java b/packages/SystemUI/src/com/android/systemui/statusbar/notification/footer/ui/view/FooterView.java
index 3caaf542e787..3f5316e11c31 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/notification/footer/ui/view/FooterView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/notification/footer/ui/view/FooterView.java
@@ -20,7 +20,6 @@ import static android.graphics.PorterDuff.Mode.SRC_ATOP;
 
 import static com.android.systemui.Flags.notificationFooterBackgroundTintOptimization;
 import static com.android.systemui.Flags.notificationShadeBlur;
-import static com.android.systemui.util.ColorUtilKt.hexColorString;
 
 import android.annotation.ColorInt;
 import android.annotation.DrawableRes;
@@ -45,14 +44,12 @@ import com.android.internal.graphics.ColorUtils;
 import com.android.systemui.common.shared.colors.SurfaceEffectColors;
 import com.android.systemui.res.R;
 import com.android.systemui.scene.shared.flag.SceneContainerFlag;
-import com.android.systemui.statusbar.notification.ColorUpdateLogger;
 import com.android.systemui.statusbar.notification.footer.shared.NotifRedesignFooter;
 import com.android.systemui.statusbar.notification.row.FooterViewButton;
 import com.android.systemui.statusbar.notification.row.StackScrollerDecorView;
 import com.android.systemui.statusbar.notification.stack.AnimationProperties;
 import com.android.systemui.statusbar.notification.stack.ExpandableViewState;
 import com.android.systemui.statusbar.notification.stack.ViewState;
-import com.android.systemui.util.DrawableDumpKt;
 import com.android.systemui.util.DumpUtilsKt;
 
 import java.io.PrintWriter;
@@ -280,10 +277,6 @@ public class FooterView extends StackScrollerDecorView {
 
     @Override
     protected void onFinishInflate() {
-        ColorUpdateLogger colorUpdateLogger = ColorUpdateLogger.getInstance();
-        if (colorUpdateLogger != null) {
-            colorUpdateLogger.logTriggerEvent("Footer.onFinishInflate()");
-        }
         super.onFinishInflate();
         mClearAllButton = (FooterViewButton) findSecondaryView();
         if (NotifRedesignFooter.isEnabled()) {
@@ -365,10 +358,6 @@ public class FooterView extends StackScrollerDecorView {
 
     @Override
     protected void onConfigurationChanged(Configuration newConfig) {
-        ColorUpdateLogger colorUpdateLogger = ColorUpdateLogger.getInstance();
-        if (colorUpdateLogger != null) {
-            colorUpdateLogger.logTriggerEvent("Footer.onConfigurationChanged()");
-        }
         super.onConfigurationChanged(newConfig);
         updateColors();
         updateContent();
@@ -418,8 +407,6 @@ public class FooterView extends StackScrollerDecorView {
                     historyBg.setColorFilter(bgColorFilter);
                 }
             }
-        } else {
-            scHigh = 0;
         }
         mClearAllButton.setBackground(clearAllBg);
         mClearAllButton.setTextColor(onSurface);
@@ -435,13 +422,6 @@ public class FooterView extends StackScrollerDecorView {
         }
         mSeenNotifsFooterTextView.setTextColor(onSurface);
         mSeenNotifsFooterTextView.setCompoundDrawableTintList(ColorStateList.valueOf(onSurface));
-        ColorUpdateLogger colorUpdateLogger = ColorUpdateLogger.getInstance();
-        if (colorUpdateLogger != null) {
-            colorUpdateLogger.logEvent("Footer.updateColors()",
-                    "textColor(onSurface)=" + hexColorString(onSurface)
-                            + " backgroundTint(surfaceContainerHigh)=" + hexColorString(scHigh)
-                            + " background=" + DrawableDumpKt.dumpToString(settingsBg));
-        }
     }
 
     public void setIsBlurSupported(boolean isBlurSupported) {
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/notification/row/ActivatableNotificationView.java b/packages/SystemUI/src/com/android/systemui/statusbar/notification/row/ActivatableNotificationView.java
index 43c09d4e806e..82e92d7ed97d 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/notification/row/ActivatableNotificationView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/notification/row/ActivatableNotificationView.java
@@ -97,7 +97,7 @@ public abstract class ActivatableNotificationView extends ExpandableOutlineView
     private boolean mActivated;
 
     private Interpolator mCurrentAppearInterpolator;
-    protected NotificationBackgroundView mBackgroundNormal;
+    NotificationBackgroundView mBackgroundNormal;
     private float mAnimationTranslationY;
     private boolean mDrawingAppearAnimation;
     private ValueAnimator mAppearAnimator;
@@ -178,10 +178,6 @@ public abstract class ActivatableNotificationView extends ExpandableOutlineView
         updateBackgroundTint();
     }
 
-    protected int getNormalBgColor() {
-        return mNormalColor;
-    }
-
     /**
      * @param width The actual width to apply to the background view.
      */
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/notification/row/ExpandableNotificationRow.java b/packages/SystemUI/src/com/android/systemui/statusbar/notification/row/ExpandableNotificationRow.java
index 8a539747fa47..8d442fc4276a 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/notification/row/ExpandableNotificationRow.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/notification/row/ExpandableNotificationRow.java
@@ -29,7 +29,6 @@ import static com.android.systemui.statusbar.notification.NotificationUtils.logK
 import static com.android.systemui.statusbar.notification.collection.NotificationEntry.DismissState.PARENT_DISMISSED;
 import static com.android.systemui.statusbar.notification.row.NotificationContentView.VISIBLE_TYPE_HEADSUP;
 import static com.android.systemui.statusbar.policy.RemoteInputView.FOCUS_ANIMATION_MIN_SCALE;
-import static com.android.systemui.util.ColorUtilKt.hexColorString;
 
 import android.animation.Animator;
 import android.animation.AnimatorListenerAdapter;
@@ -108,7 +107,6 @@ import com.android.systemui.statusbar.SmartReplyController;
 import com.android.systemui.statusbar.StatusBarIconView;
 import com.android.systemui.statusbar.notification.AboveShelfChangedListener;
 import com.android.systemui.statusbar.notification.BundleInteractionLogger;
-import com.android.systemui.statusbar.notification.ColorUpdateLogger;
 import com.android.systemui.statusbar.notification.FeedbackIcon;
 import com.android.systemui.statusbar.notification.LaunchAnimationParameters;
 import com.android.systemui.statusbar.notification.NmSummarizationUiFlag;
@@ -216,7 +214,6 @@ public class ExpandableNotificationRow extends ActivatableNotificationView
     private NotificationActivityStarter mNotificationActivityStarter;
     private MetricsLogger mMetricsLogger;
     private NotificationChildrenContainerLogger mChildrenContainerLogger;
-    private ColorUpdateLogger mColorUpdateLogger;
     private BundleInteractionLogger mBundleInteractionLogger;
     private NotificationDismissibilityProvider mDismissibilityProvider;
     private int mIconTransformContentShift;
@@ -573,7 +570,6 @@ public class ExpandableNotificationRow extends ActivatableNotificationView
 
     /**
      * Sets animations running in the layouts of this row, including public, private, and children.
-     *
      * @param running whether the animations should be started running or stopped.
      */
     public void setAnimationRunning(boolean running) {
@@ -730,12 +726,6 @@ public class ExpandableNotificationRow extends ActivatableNotificationView
 
     public void updateBackgroundColorsOfSelf() {
         super.updateBackgroundColors();
-        if (mColorUpdateLogger.isEnabled()) {
-            mColorUpdateLogger.logNotificationEvent("ENR.updateBackgroundColorsOfSelf()",
-                    mLoggingKey,
-                    "normalBgColor=" + hexColorString(getNormalBgColor())
-                            + " background=" + mBackgroundNormal.toDumpString());
-        }
     }
 
     @Override
@@ -1746,7 +1736,7 @@ public class ExpandableNotificationRow extends ActivatableNotificationView
     }
 
     public void setContentBackground(int customBackgroundColor, boolean animate,
-            NotificationContentView notificationContentView) {
+                                     NotificationContentView notificationContentView) {
         if (getShowingLayout() == notificationContentView) {
             setTintColor(customBackgroundColor, animate);
         }
@@ -1826,7 +1816,7 @@ public class ExpandableNotificationRow extends ActivatableNotificationView
     }
 
     /**
-     * @return if this entry should be kept in its parent during removal.
+     * @return  if this entry should be kept in its parent during removal.
      */
     public boolean keepInParentForDismissAnimation() {
         return mKeepInParentForDismissAnimation;
@@ -2270,7 +2260,6 @@ public class ExpandableNotificationRow extends ActivatableNotificationView
             NotificationDismissibilityProvider dismissibilityProvider,
             MetricsLogger metricsLogger,
             NotificationChildrenContainerLogger childrenContainerLogger,
-            ColorUpdateLogger colorUpdateLogger,
             SmartReplyConstants smartReplyConstants,
             SmartReplyController smartReplyController,
             IStatusBarService statusBarService,
@@ -2331,7 +2320,6 @@ public class ExpandableNotificationRow extends ActivatableNotificationView
         mNotificationGutsManager = gutsManager;
         mMetricsLogger = metricsLogger;
         mChildrenContainerLogger = childrenContainerLogger;
-        mColorUpdateLogger = colorUpdateLogger;
         mDismissibilityProvider = dismissibilityProvider;
         mBundleInteractionLogger = bundleInteractionLogger;
         setHapticFeedbackEnabled(!Flags.msdlFeedback());
@@ -2868,7 +2856,7 @@ public class ExpandableNotificationRow extends ActivatableNotificationView
     }
 
     public Animator getTranslateViewAnimator(final float leftTarget,
-            AnimatorUpdateListener listener) {
+                                             AnimatorUpdateListener listener) {
         if (mTranslateAnim != null) {
             mTranslateAnim.cancel();
         }
@@ -3346,7 +3334,6 @@ public class ExpandableNotificationRow extends ActivatableNotificationView
         }
         return getCollapsedHeight();
     }
-
     /**
      * @return {@code true} if the notification can show it's heads up layout. This is mostly true
      * except for legacy use cases.
@@ -3632,7 +3619,7 @@ public class ExpandableNotificationRow extends ActivatableNotificationView
 
     @Override
     public void setHideSensitive(boolean hideSensitive, boolean animated, long delay,
-            long duration) {
+                                 long duration) {
         if (getVisibility() == GONE) {
             // If we are GONE, the hideSensitive parameter will not be calculated and always be
             // false, which is incorrect, let's wait until a real call comes in later.
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/notification/row/ExpandableNotificationRowController.java b/packages/SystemUI/src/com/android/systemui/statusbar/notification/row/ExpandableNotificationRowController.java
index 4c168b947934..90364031c4fc 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/notification/row/ExpandableNotificationRowController.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/notification/row/ExpandableNotificationRowController.java
@@ -49,7 +49,6 @@ import com.android.systemui.plugins.statusbar.StatusBarStateController;
 import com.android.systemui.scene.shared.flag.SceneContainerFlag;
 import com.android.systemui.statusbar.SmartReplyController;
 import com.android.systemui.statusbar.notification.BundleInteractionLogger;
-import com.android.systemui.statusbar.notification.ColorUpdateLogger;
 import com.android.systemui.statusbar.notification.FeedbackIcon;
 import com.android.systemui.statusbar.notification.NotificationActivityStarter;
 import com.android.systemui.statusbar.notification.collection.EntryAdapterFactory;
@@ -98,7 +97,6 @@ public class ExpandableNotificationRowController implements NotifViewController
     private final ActivatableNotificationViewController mActivatableNotificationViewController;
     private final PluginManager mPluginManager;
     private final SystemClock mClock;
-    private final ColorUpdateLogger mColorUpdateLogger;
     private final KeyguardBypassController mKeyguardBypassController;
     private final GroupMembershipManager mGroupMembershipManager;
     private final GroupExpansionManager mGroupExpansionManager;
@@ -265,7 +263,6 @@ public class ExpandableNotificationRowController implements NotifViewController
             ActivatableNotificationViewController activatableNotificationViewController,
             RemoteInputViewSubcomponent.Factory rivSubcomponentFactory,
             MetricsLogger metricsLogger,
-            ColorUpdateLogger colorUpdateLogger,
             NotificationRowLogger logBufferLogger,
             NotificationChildrenContainerLogger childrenContainerLogger,
             NotificationListContainer listContainer,
@@ -326,7 +323,6 @@ public class ExpandableNotificationRowController implements NotifViewController
         mDragController = dragController;
         mMetricsLogger = metricsLogger;
         mChildrenContainerLogger = childrenContainerLogger;
-        mColorUpdateLogger = colorUpdateLogger;
         mLogBufferLogger = logBufferLogger;
         mSmartReplyConstants = smartReplyConstants;
         mSmartReplyController = smartReplyController;
@@ -397,7 +393,6 @@ public class ExpandableNotificationRowController implements NotifViewController
                 mDismissibilityProvider,
                 mMetricsLogger,
                 mChildrenContainerLogger,
-                mColorUpdateLogger,
                 mSmartReplyConstants,
                 mSmartReplyController,
                 mStatusBarService,
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/notification/row/ExpandableView.java b/packages/SystemUI/src/com/android/systemui/statusbar/notification/row/ExpandableView.java
index c715add62bee..c3edd3b5ab8b 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/notification/row/ExpandableView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/notification/row/ExpandableView.java
@@ -16,7 +16,6 @@
 
 package com.android.systemui.statusbar.notification.row;
 
-import static com.android.systemui.Flags.notificationColorUpdateLogger;
 import static com.android.systemui.Flags.physicalNotificationMovement;
 
 import android.animation.AnimatorListenerAdapter;
@@ -77,8 +76,8 @@ public abstract class ExpandableView extends FrameLayout implements Dumpable, Ro
             });
     private static final String TAG = "ExpandableView";
     /** whether the dump() for this class should include verbose details */
-    protected static final boolean DUMP_VERBOSE = Compile.IS_DEBUG
-            && (Log.isLoggable(TAG, Log.VERBOSE) || notificationColorUpdateLogger());
+    protected static final boolean DUMP_VERBOSE =
+            Compile.IS_DEBUG && Log.isLoggable(TAG, Log.VERBOSE);
 
     private RoundableState mRoundableState = null;
     protected OnHeightChangedListener mOnHeightChangedListener;
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/notification/row/NotificationBackgroundView.java b/packages/SystemUI/src/com/android/systemui/statusbar/notification/row/NotificationBackgroundView.java
index a1987581fab3..e9ac11c0c3c0 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/notification/row/NotificationBackgroundView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/notification/row/NotificationBackgroundView.java
@@ -41,7 +41,6 @@ import com.android.systemui.Dumpable;
 import com.android.systemui.common.shared.colors.SurfaceEffectColors;
 import com.android.systemui.res.R;
 import com.android.systemui.statusbar.notification.shared.NotificationAddXOnHoverToDismiss;
-import com.android.systemui.util.DrawableDumpKt;
 
 import java.io.PrintWriter;
 import java.util.Arrays;
@@ -496,16 +495,6 @@ public class NotificationBackgroundView extends View implements Dumpable,
         pw.println("mActualHeight: " + mActualHeight);
         pw.println("mTintColor: " + hexColorString(mTintColor));
         pw.println("mRippleColor: " + hexColorString(mRippleColor));
-        pw.println("mBackground: " + DrawableDumpKt.dumpToString(mBackground));
-    }
-
-    /** create a concise dump of this view's colors */
-    public String toDumpString() {
-        return "<NotificationBackgroundView"
-                + " tintColor=" + hexColorString(mTintColor)
-                + " rippleColor=" + hexColorString(mRippleColor)
-                + " bgColor=" + DrawableDumpKt.getSolidColor(mBackground)
-                + ">";
-
+        pw.println("mBackground: " + mBackground);
     }
 }
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/notification/stack/NotificationStackScrollLayout.java b/packages/SystemUI/src/com/android/systemui/statusbar/notification/stack/NotificationStackScrollLayout.java
index 8ce2b5f637a1..0ef56b344556 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/notification/stack/NotificationStackScrollLayout.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/notification/stack/NotificationStackScrollLayout.java
@@ -103,7 +103,6 @@ import com.android.systemui.shade.TouchLogger;
 import com.android.systemui.statusbar.NotificationShelf;
 import com.android.systemui.statusbar.StatusBarState;
 import com.android.systemui.statusbar.headsup.shared.StatusBarNoHunBehavior;
-import com.android.systemui.statusbar.notification.ColorUpdateLogger;
 import com.android.systemui.statusbar.notification.FakeShadowView;
 import com.android.systemui.statusbar.notification.LaunchAnimationParameters;
 import com.android.systemui.statusbar.notification.NotificationTransitionAnimatorController;
@@ -807,8 +806,8 @@ public class NotificationStackScrollLayout
     void updateBgColor() {
         for (int i = 0; i < getChildCount(); i++) {
             View child = getChildAt(i);
-            if (child instanceof ActivatableNotificationView activatableView) {
-                activatableView.updateBackgroundColors();
+            if (child instanceof ActivatableNotificationView) {
+                ((ActivatableNotificationView) child).updateBackgroundColors();
             }
         }
     }
@@ -5140,13 +5139,6 @@ public class NotificationStackScrollLayout
         final @ColorInt int onSurfaceVariant = mContext.getColor(
                 com.android.internal.R.color.materialColorOnSurfaceVariant);
 
-        ColorUpdateLogger colorUpdateLogger = ColorUpdateLogger.getInstance();
-        if (colorUpdateLogger != null) {
-            colorUpdateLogger.logEvent("NSSL.updateDecorViews()",
-                    "onSurface=" + ColorUtilKt.hexColorString(onSurface)
-                            + " onSurfaceVariant=" + ColorUtilKt.hexColorString(onSurfaceVariant));
-        }
-
         mSectionsManager.setHeaderForegroundColors(onSurface, onSurfaceVariant);
 
         if (mFooterView != null) {
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/notification/stack/NotificationStackScrollLayoutController.java b/packages/SystemUI/src/com/android/systemui/statusbar/notification/stack/NotificationStackScrollLayoutController.java
index ff5ad1d32abe..3c9f63f62ac3 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/notification/stack/NotificationStackScrollLayoutController.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/notification/stack/NotificationStackScrollLayoutController.java
@@ -93,7 +93,6 @@ import com.android.systemui.statusbar.NotificationShelf;
 import com.android.systemui.statusbar.RemoteInputController;
 import com.android.systemui.statusbar.StatusBarState;
 import com.android.systemui.statusbar.SysuiStatusBarStateController;
-import com.android.systemui.statusbar.notification.ColorUpdateLogger;
 import com.android.systemui.statusbar.notification.DynamicPrivacyController;
 import com.android.systemui.statusbar.notification.LaunchAnimationParameters;
 import com.android.systemui.statusbar.notification.NotificationWakeUpCoordinator;
@@ -171,7 +170,6 @@ public class NotificationStackScrollLayoutController implements Dumpable {
     private final DynamicPrivacyController mDynamicPrivacyController;
     private final ConfigurationController mConfigurationController;
     private final MetricsLogger mMetricsLogger;
-    private final ColorUpdateLogger mColorUpdateLogger;
 
     private final DumpManager mDumpManager;
     private final FalsingCollector mFalsingCollector;
@@ -222,7 +220,6 @@ public class NotificationStackScrollLayoutController implements Dumpable {
             new View.OnAttachStateChangeListener() {
                 @Override
                 public void onViewAttachedToWindow(View v) {
-                    mColorUpdateLogger.logTriggerEvent("NSSLC.onViewAttachedToWindow()");
                     mConfigurationController.addCallback(mConfigurationListener);
                     final int newBarState = mStatusBarStateController.getState();
                     if (newBarState != mBarState) {
@@ -235,7 +232,6 @@ public class NotificationStackScrollLayoutController implements Dumpable {
 
                 @Override
                 public void onViewDetachedFromWindow(View v) {
-                    mColorUpdateLogger.logTriggerEvent("NSSLC.onViewDetachedFromWindow()");
                     mConfigurationController.removeCallback(mConfigurationListener);
                     mStatusBarStateController.removeCallback(mStateListener);
                 }
@@ -292,16 +288,12 @@ public class NotificationStackScrollLayoutController implements Dumpable {
 
         @Override
         public void onUiModeChanged() {
-            mColorUpdateLogger.logTriggerEvent("NSSLC.onUiModeChanged()",
-                    "mode=" + mConfigurationController.getNightModeName());
             mView.updateBgColor();
             mView.updateDecorViews();
         }
 
         @Override
         public void onThemeChanged() {
-            mColorUpdateLogger.logTriggerEvent("NSSLC.onThemeChanged()",
-                    "mode=" + mConfigurationController.getNightModeName());
             mView.updateCornerRadius();
             mView.updateBgColor();
             mView.updateDecorViews();
@@ -854,7 +846,6 @@ public class NotificationStackScrollLayoutController implements Dumpable {
             PowerInteractor powerInteractor,
             NotificationLockscreenUserManager lockscreenUserManager,
             MetricsLogger metricsLogger,
-            ColorUpdateLogger colorUpdateLogger,
             DumpManager dumpManager,
             FalsingCollector falsingCollector,
             FalsingManager falsingManager,
@@ -908,7 +899,6 @@ public class NotificationStackScrollLayoutController implements Dumpable {
         mPowerInteractor = powerInteractor;
         mLockscreenUserManager = lockscreenUserManager;
         mMetricsLogger = metricsLogger;
-        mColorUpdateLogger = colorUpdateLogger;
         mDumpManager = dumpManager;
         mLockscreenShadeTransitionController = lockscreenShadeTransitionController;
         mFalsingCollector = falsingCollector;
diff --git a/packages/SystemUI/src/com/android/systemui/util/DrawableDump.kt b/packages/SystemUI/src/com/android/systemui/util/DrawableDump.kt
deleted file mode 100644
index 0c079a309962..000000000000
--- a/packages/SystemUI/src/com/android/systemui/util/DrawableDump.kt
+++ /dev/null
@@ -1,201 +0,0 @@
-/*
- * Copyright (C) 2023 The Android Open Source Project
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package com.android.systemui.util
-
-import android.content.res.ColorStateList
-import android.graphics.BlendMode
-import android.graphics.BlendModeColorFilter
-import android.graphics.ColorFilter
-import android.graphics.LightingColorFilter
-import android.graphics.PorterDuffColorFilter
-import android.graphics.drawable.Drawable
-import android.graphics.drawable.DrawableWrapper
-import android.graphics.drawable.GradientDrawable
-import android.graphics.drawable.LayerDrawable
-import android.graphics.drawable.RippleDrawable
-import android.util.Log
-
-fun dumpToString(drawable: Drawable?): String =
-    if (Compile.IS_DEBUG) StringBuilder().appendDrawable(drawable).toString()
-    else drawable.toString()
-
-fun getSolidColor(drawable: Drawable?): String =
-    if (Compile.IS_DEBUG) hexColorString(getSolidColors(drawable)?.defaultColor)
-    else if (drawable == null) "null" else "?"
-
-private fun getSolidColors(drawable: Drawable?): ColorStateList? {
-    return when (drawable) {
-        is GradientDrawable -> {
-            return drawable.getStateField<ColorStateList>("mSolidColors")
-        }
-        is LayerDrawable -> {
-            for (iLayer in 0 until drawable.numberOfLayers) {
-                getSolidColors(drawable.getDrawable(iLayer))?.let {
-                    return it
-                }
-            }
-            null
-        }
-        is DrawableWrapper -> {
-            return getSolidColors(drawable.drawable)
-        }
-        else -> null
-    }
-}
-
-private fun StringBuilder.appendDrawable(drawable: Drawable?): StringBuilder {
-    if (drawable == null) {
-        append("null")
-        return this
-    }
-    append("<")
-    append(drawable.javaClass.simpleName)
-
-    drawable.getStateField<ColorStateList>("mTint", fieldRequired = false)?.let {
-        append(" tint=")
-        appendColors(it)
-        append(" blendMode=")
-        append(drawable.getStateField<BlendMode>("mBlendMode"))
-    }
-    drawable.colorFilter
-        ?.takeUnless { drawable is DrawableWrapper }
-        ?.let {
-            append(" colorFilter=")
-            appendColorFilter(it)
-        }
-    when (drawable) {
-        is DrawableWrapper -> {
-            append(" wrapped=")
-            appendDrawable(drawable.drawable)
-        }
-        is LayerDrawable -> {
-            if (drawable is RippleDrawable) {
-                drawable.getStateField<ColorStateList>("mColor")?.let {
-                    append(" color=")
-                    appendColors(it)
-                }
-                drawable.effectColor?.let {
-                    append(" effectColor=")
-                    appendColors(it)
-                }
-            }
-            append(" layers=[")
-            for (iLayer in 0 until drawable.numberOfLayers) {
-                if (iLayer != 0) append(", ")
-                appendDrawable(drawable.getDrawable(iLayer))
-            }
-            append("]")
-        }
-        is GradientDrawable -> {
-            drawable
-                .getStateField<Int>("mShape")
-                ?.takeIf { it != 0 }
-                ?.let {
-                    append(" shape=")
-                    append(it)
-                }
-            drawable.getStateField<ColorStateList>("mSolidColors")?.let {
-                append(" solidColors=")
-                appendColors(it)
-            }
-            drawable.getStateField<ColorStateList>("mStrokeColors")?.let {
-                append(" strokeColors=")
-                appendColors(it)
-            }
-            drawable.colors?.let {
-                append(" gradientColors=[")
-                it.forEachIndexed { iColor, color ->
-                    if (iColor != 0) append(", ")
-                    append(hexColorString(color))
-                }
-                append("]")
-            }
-        }
-    }
-    append(">")
-    return this
-}
-
-private inline fun <reified T> Drawable.getStateField(
-    name: String,
-    fieldRequired: Boolean = true
-): T? {
-    val state = this.constantState ?: return null
-    val clazz = state.javaClass
-    return try {
-        val field = clazz.getDeclaredField(name)
-        field.isAccessible = true
-        field.get(state) as T?
-    } catch (ex: Exception) {
-        if (fieldRequired) {
-            Log.w(TAG, "Missing ${clazz.simpleName}.$name: ${T::class.simpleName}", ex)
-        } else if (Log.isLoggable(TAG, Log.VERBOSE)) {
-            Log.v(TAG, "Missing ${clazz.simpleName}.$name: ${T::class.simpleName} ($ex)")
-        }
-        null
-    }
-}
-
-private fun Appendable.appendColors(colorStateList: ColorStateList?) {
-    if (colorStateList == null) {
-        append("null")
-        return
-    }
-    val colors = colorStateList.colors
-    if (colors.size == 1) {
-        append(hexColorString(colors[0]))
-        return
-    }
-    append("<ColorStateList size=")
-    append(colors.size.toString())
-    append(" default=")
-    append(hexColorString(colorStateList.defaultColor))
-    append(">")
-}
-
-private fun Appendable.appendColorFilter(colorFilter: ColorFilter?) {
-    if (colorFilter == null) {
-        append("null")
-        return
-    }
-    append("<")
-    append(colorFilter.javaClass.simpleName)
-    when (colorFilter) {
-        is PorterDuffColorFilter -> {
-            append(" color=")
-            append(hexColorString(colorFilter.color))
-            append(" mode=")
-            append(colorFilter.mode.toString())
-        }
-        is BlendModeColorFilter -> {
-            append(" color=")
-            append(hexColorString(colorFilter.color))
-            append(" mode=")
-            append(colorFilter.mode.toString())
-        }
-        is LightingColorFilter -> {
-            append(" multiply=")
-            append(hexColorString(colorFilter.colorMultiply))
-            append(" add=")
-            append(hexColorString(colorFilter.colorAdd))
-        }
-        else -> append(" unhandled")
-    }
-    append(">")
-}
-
-private const val TAG = "DrawableDump"
diff --git a/packages/SystemUI/tests/src/com/android/systemui/statusbar/notification/collection/coordinator/ViewConfigCoordinatorTest.kt b/packages/SystemUI/tests/src/com/android/systemui/statusbar/notification/collection/coordinator/ViewConfigCoordinatorTest.kt
index ff17a362eb32..5c93e47da836 100644
--- a/packages/SystemUI/tests/src/com/android/systemui/statusbar/notification/collection/coordinator/ViewConfigCoordinatorTest.kt
+++ b/packages/SystemUI/tests/src/com/android/systemui/statusbar/notification/collection/coordinator/ViewConfigCoordinatorTest.kt
@@ -23,7 +23,6 @@ import com.android.keyguard.KeyguardUpdateMonitorCallback
 import com.android.systemui.SysuiTestCase
 import com.android.systemui.statusbar.NotificationLockscreenUserManager
 import com.android.systemui.statusbar.NotificationLockscreenUserManager.UserChangedListener
-import com.android.systemui.statusbar.notification.ColorUpdateLogger
 import com.android.systemui.statusbar.notification.collection.NotifPipeline
 import com.android.systemui.statusbar.notification.collection.NotificationEntry
 import com.android.systemui.statusbar.notification.row.ExpandableNotificationRow
@@ -58,7 +57,6 @@ class ViewConfigCoordinatorTest : SysuiTestCase() {
     private val lockscreenUserManager: NotificationLockscreenUserManager = mock()
     private val gutsManager: NotificationGutsManager = mock()
     private val keyguardUpdateMonitor: KeyguardUpdateMonitor = mock()
-    private val colorUpdateLogger: ColorUpdateLogger = mock()
 
     @Before
     fun setUp() {
@@ -70,7 +68,6 @@ class ViewConfigCoordinatorTest : SysuiTestCase() {
                 lockscreenUserManager,
                 gutsManager,
                 keyguardUpdateMonitor,
-                colorUpdateLogger,
             )
         coordinator.attach(pipeline)
         userChangedListener = withArgCaptor {
diff --git a/packages/SystemUI/tests/utils/src/com/android/systemui/statusbar/notification/row/ExpandableNotificationRowBuilder.kt b/packages/SystemUI/tests/utils/src/com/android/systemui/statusbar/notification/row/ExpandableNotificationRowBuilder.kt
index 84a5e9914485..2c627dad33c4 100644
--- a/packages/SystemUI/tests/utils/src/com/android/systemui/statusbar/notification/row/ExpandableNotificationRowBuilder.kt
+++ b/packages/SystemUI/tests/utils/src/com/android/systemui/statusbar/notification/row/ExpandableNotificationRowBuilder.kt
@@ -57,7 +57,6 @@ import com.android.systemui.statusbar.NotificationShadeWindowController
 import com.android.systemui.statusbar.RankingBuilder
 import com.android.systemui.statusbar.SmartReplyController
 import com.android.systemui.statusbar.notification.BundleInteractionLogger
-import com.android.systemui.statusbar.notification.ColorUpdateLogger
 import com.android.systemui.statusbar.notification.ConversationNotificationManager
 import com.android.systemui.statusbar.notification.ConversationNotificationProcessor
 import com.android.systemui.statusbar.notification.NotificationActivityStarter
@@ -454,7 +453,6 @@ class ExpandableNotificationRowBuilder(
             mDismissibilityProvider,
             Mockito.mock(MetricsLogger::class.java, STUB_ONLY),
             Mockito.mock(NotificationChildrenContainerLogger::class.java, STUB_ONLY),
-            Mockito.mock(ColorUpdateLogger::class.java, STUB_ONLY),
             mSmartReplyConstants,
             mSmartReplyController,
             Mockito.mock(IStatusBarService::class.java, STUB_ONLY),
-- 
2.43.0

